core/src/test/java/org/elasticsearch/action/support/master/IndexingMasterFailoverIT.java

```diff
@@ -73,9 +73,10 @@ public void testMasterFailoverDuringIndexingWithMappingChanges() throws Throwabl
                 .put("discovery.zen.join_timeout", "10s")  // still long to induce failures but to long so test won't time out
                 .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "1s") // <-- for hitting simulated network failures quickly
                 .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 2)
+                .put(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.getKey(), "0")  // <-- we wait for cluster formation at the end
```

Comment:
> No. It's used as an internal setting to start up tribe node

Just an FYI  - it's that setting has been there at least since 0.90.

> I think that this is the wrong approach and that there is a cleaner way to do this.